llvm/lib/IR/DebugProgramInstruction.cpp

```diff
@@ -267,6 +267,19 @@ void DPMarker::dropOneDPValue(DPValue *DPV) {
   DPV->deleteInstr();
 }
 
+std::optional<DPValue::self_iterator> DPMarker::getReinsertionPosition() {
```

Comment:
Does this need to be a `DPMarker` method? It doesn't look like any members are used other than `MarkedInstr`, which I imagine can be found easily at call sites.

The use-case in #73660 looks like it creates a DPMarker just to call this function (I could be misreading it though?).

Reply:
Uuuuhhhh. Yeah, I think it does.

It's really not clear where this method should be placed -- I think it was in originally in BasicBlock, then I moved it here.... how would you feel about it being back in BasicBlock?

Reply:
Yeah that sounds ok to me. Or can it be standalone (if so, maybe a static method on DPMarker makes sense?)? Unsure.